WordPress/Classes/Services/PostService.m

```diff
@@ -824,7 +827,7 @@ - (void)updatePost:(AbstractPost *)post withRemotePost:(RemotePost *)remotePost
     if (post.pathForDisplayImage.length == 0) {
         [post updatePathForDisplayImageBasedOnContent];
     }
-    post.authorAvatarURL = remotePost.authorAvatarURL;
+    post.authorAvatarURL = remotePost.authorAvatarURL ?: author.avatarURL;
```

Comment:
Could you please expand on this change?
It looks like what this does is use the `avatarURL` from the `Blog.authors` instead of the remote post. Had you observed an issue with avatars, similar to the issue with author names, that led you to make this change?

Reply:
Glad to!

> It looks like what this does is use the avatarURL from the Blog.authors instead of the remote post.

Correct - it will try to use a queried author's avatarURL in the case that the remotePost didn't include it. 

In my testing of this, WordPress.com sites included an author on the remote post ([see the author object response parameter here](https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/posts/%24post_ID/)), so `remotePost.authorAvatarURL` was always available.

Self-hosted sites, however, did not include an author object so the the author query was introduced.

Though `author.avatarURL` is not used for the feature to set and change authors, I set the property because I felt it made the solution more complete (e.g. if we have an author already, let's set the avatarURL instead of `nil`).

Please let me know if I should clarify any of this! Thanks.

Reply:
That makes sense. Looks like the XML-RPC API only provides the post's author ID so it makes sense to use the author object from the Blog to fetch these extra attributes like avatar URL.